File: Premtek.Base/CLaserReader_KeyenceILS065Voltage.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
using ProjectCore;
//using ProjectIO;
namespace Premtek.Base
{

	public class CLaserReader_KeyenceILS065Voltage : ILaserReader
	{


		public bool ChangeProgram(int ProgramID)
		{
            MDateLog.gSyslog.Save("ChangeProgram Function Not Supported.");
			return true;
		}

		public bool Close()
		{
            MDateLog.gSyslog.Save("ChangeProgram Function Not Supported.");
			return true;
		}


		public bool EthernetOpen(string IP, int Port)
		{
            MDateLog.gSyslog.Save("EthernetOpen Function Not Supported.");
			return true;
		}

		public bool GetValue(string Mode, ref string value, int aiIndex = 0, bool waitReturn = false)
		{
            //TODO: 專案相依性異常 合併後再調整
            //value = MCommonIO.gAICollection.Value[aiIndex].ToString(); 
			return true;
		}

		public bool GetVersion(ref string Version)
		{
            MDateLog.gSyslog.Save("GetVersion Function Not Supported.");
			return true;
		}

		public bool RebootController()
		{
            MDateLog.gSyslog.Save("RebootController Function Not Supported.");
			return true;
		}

		public bool IsTimeOut {
			get { return false; }
		}

		public bool PortIsOpen {
			get { return true; }
		}

		public int TimeoutTimer { get; set; }

		public bool IsBusy {
			get { return false; }
		}

        bool ILaserReader.IsTimeOut
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        int ILaserReader.TimeoutTimer
        {
            get
            {
                throw new NotImplementedException();
            }

            set
            {
                throw new NotImplementedException();
            }
        }

        bool ILaserReader.PortIsOpen
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        bool ILaserReader.IsBusy
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        MCommonDefine.sReceiveStatus ILaserReader.Result
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public bool Initial(string PortName, string BaudRate)
		{
			MDateLog.gSyslog.Save("Initial Function is Not Supported at CLaserReader_KeyenceILS065Voltage.");
			return false;
		}

        //Public Sub ResetState() Implements ILaserReader.ResetState

        //End Sub
        /// <summary>接收資料結果</summary>
        /// <remarks></remarks>
        MCommonDefine.sReceiveStatus[] mResult = new MCommonDefine.sReceiveStatus[4];
		public MCommonDefine.sReceiveStatus Result(int channelNo) {
			 return mResult[channelNo];
		}

        bool ILaserReader.EthernetOpen(string IP, int Port)
        {
            throw new NotImplementedException();
        }

        bool ILaserReader.Initial(string PortName, string BaudRate)
        {
            throw new NotImplementedException();
        }

        bool ILaserReader.Close()
        {
            throw new NotImplementedException();
        }

        bool ILaserReader.GetValue(string Mode, ref string data, int aiIndex, bool waitReturn)
        {
            throw new NotImplementedException();
        }

        bool ILaserReader.GetVersion(ref string Version)
        {
            throw new NotImplementedException();
        }

        bool ILaserReader.RebootController()
        {
            throw new NotImplementedException();
        }

        bool ILaserReader.ChangeProgram(int ProgramID)
        {
            throw new NotImplementedException();
        }
    }
}

File: ProjectLaserInterferometer/CLaserReader_KeyenceILS065Voltage.vb
Imports ProjectCore
Imports ProjectIO

Public Class CLaserReader_KeyenceILS065Voltage
    Implements ILaserReader


    Public Function ChangeProgram(ProgramID As Integer) As Boolean Implements ILaserReader.ChangeProgram
        gSyslog.Save("ChangeProgram Function Not Supported.")
        Return True
    End Function

    Public Function Close() As Boolean Implements ILaserReader.Close
        gSyslog.Save("ChangeProgram Function Not Supported.")
        Return True
    End Function


    Public Function EthernetOpen(IP As String, Port As Integer) As Boolean Implements ILaserReader.EthernetOpen
        gSyslog.Save("EthernetOpen Function Not Supported.")
        Return True
    End Function

    Public Function GetValue(ByVal Mode As String, ByRef value As String, Optional aiIndex As Integer = 0, Optional ByVal waitReturn As Boolean = False) As Boolean Implements ILaserReader.GetValue
        value = gAICollection.Value(aiIndex)
        Return True
    End Function

    Public Function GetVersion(ByRef Version As String) As Boolean Implements ILaserReader.GetVersion
        gSyslog.Save("GetVersion Function Not Supported.")
        Return True
    End Function

    Public Function RebootController() As Boolean Implements ILaserReader.RebootController
        gSyslog.Save("RebootController Function Not Supported.")
        Return True
    End Function

    Public ReadOnly Property IsTimeOut As Boolean Implements ILaserReader.IsTimeOut
        Get
            Return False
        End Get
    End Property

    Public ReadOnly Property PortIsOpen As Boolean Implements ILaserReader.PortIsOpen
        Get
            Return True
        End Get
    End Property

    Public Property TimeoutTimer As Integer Implements ILaserReader.TimeoutTimer

    Public ReadOnly Property IsBusy As Boolean Implements ILaserReader.IsBusy
        Get
            Return False
        End Get
    End Property

    Public Function Initial(PortName As String, BaudRate As String) As Boolean Implements ILaserReader.Initial
        gSyslog.Save("Initial Function is Not Supported at CLaserReader_KeyenceILS065Voltage.")
        Return False
    End Function

    'Public Sub ResetState() Implements ILaserReader.ResetState

    'End Sub
    ''' <summary>接收資料結果</summary>
    ''' <remarks></remarks>
    Dim mResult(3) As sReceiveStatus
    Public ReadOnly Property Result(ByVal channelNo As Integer) As sReceiveStatus Implements ILaserReader.Result
        Get
            Return mResult(channelNo)
        End Get
    End Property
End Class
